### PR TITLE
Card/2267/1

### PR DESCRIPTION
**Pull request by oliveiraigorm**
New time picker for the calendar.
Some details are not exactly like figma, like the footer buttons and the AM/PM button positions (due to some scrolling bugs)

image
image

### CODE REVIEW

File: src/components/Shared/DatePicker.vue [hunk 1]
```diff
@@ -403,3 +403,135 @@ export default {
...
+		}
+
+		li.active,
+		.mx-time-item:hover {
+			border-radius: var(--adminly-border-radius-button);
+		}
+	}
+
+	.mx-btn {
+		opacity: 1 !important;
```
Comment (mwalbeck) [inline src/components/Shared/DatePicker.vue:484]: To better prepare for generalised button styling I would set `margin: 0;` here and then use flexbox for aligning the buttons instead.

File: src/components/Shared/DatePicker.vue [hunk 2]
```diff
@@ -403,3 +403,135 @@ export default {
...
+<style lang="scss">
+.mx-datepicker-main{
+	box-shadow: var(--adminly-box-shadow) !important;
+	border-radius: 10px;
+	border: none !important;
+
+	.mx-datepicker-footer{
+		padding: 1rem !important;
+		border: none !important;
+		display: flex;
```
Comment (mwalbeck) [inline src/components/Shared/DatePicker.vue:415]: Add a `justify-content: space-around;` here

File: src/components/Shared/DatePicker.vue [hunk 3]
```diff
@@ -403,3 +403,135 @@ export default {
 	},
 }
 </script>
+<style lang="scss">
+.mx-datepicker-main{
+	box-shadow: var(--adminly-box-shadow) !important;
+	border-radius: 10px;
+	border: none !important;
+
```
Comment (mwalbeck) [inline src/components/Shared/DatePicker.vue:411]: I would add the following two indented rules here:
```
.mx-calendar-content,
.mx-time-content {
	height: 266px;
}

.mx-calendar-content .mx-date-row {
	margin: 3px 0;
}
```

**Review by mwalbeck — CHANGES_REQUESTED** [review]
Why is the rule to increase the width of the datepicker in adminly_core instead of here?

And also can we increase the width of it to `278px`

**Comment by oliveiraigorm** [comment]
> Why is the rule to increase the width of the datepicker in adminly_core instead of here?

There was an existing rule there but I can move it here if it makes more sense

File: src/components/Shared/DatePicker.vue (hunk 3 above)
Comment (oliveiraigorm) [reply src/components/Shared/DatePicker.vue:411]: they are causing the AM/PM to be cropped 
image

Comment (mwalbeck) [reply src/components/Shared/DatePicker.vue:411]: I can't reproduce it on my setup, is it possible something else is affecting it?

File: src/components/Shared/DatePicker.vue (hunk 1 above)
Comment (mwalbeck) [reply src/components/Shared/DatePicker.vue:484]: Sorry I found a better place to put this rule so remove it from here and then see relevant new comment for new location

File: src/components/Shared/DatePicker.vue [hunk 4]
```diff
@@ -403,3 +403,135 @@ export default {
...
+		.mx-btn.mx-datepicker-btn-confirm {
+			color: white;
+			border: none;
+			background-color: var(--adminly-blue);
+		}
+
+		.mx-btn {
+			box-shadow: var(--adminly-box-shadow);
+			border-radius: var(--adminly-border-radius-button) !important;
+			font-family: "Roc Grotesk", var(--font-face);
```
Comment (mwalbeck) [inline src/components/Shared/DatePicker.vue:426]: New location for `margin: 0;`

**Comment by mwalbeck** [comment]
> > Why is the rule to increase the width of the datepicker in adminly_core instead of here?
> 
> There was an existing rule there but I can move it here if it makes more sense

Ahh okay, but yeah I think it makes more sense to more it here

File: src/components/Shared/DatePicker.vue (hunk 3 above)
Comment (oliveiraigorm) [reply src/components/Shared/DatePicker.vue:411]: I've found a fix for it, let me know if it still works fine there for you

Comment (mwalbeck) [reply src/components/Shared/DatePicker.vue:411]: Still works here! It's a chrome specific issue, on firefox the margin-right stays `0` but on chrome it changes to `-12px` when the height changes. Very strange

**Review by mwalbeck — APPROVED** [review]

**Pull request merged**